bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java

```diff
@@ -0,0 +1,815 @@
+                                processType(decObj, "AlarmForwarding", "alarmForwarding", "boolean");
+                                processType(decObj, "usercalib", "userCalib", "calib");
+                                processType(decObj, "InitialCalFinished", "initialCalFinished", "boolean");
+                                processType(decObj, "Averaging", "averaging", "boolean");
+                                processType(decObj, "SensorInfo", "sensorInfo", "property");
+                                processType(decObj, "ErrorBars", "errorBars", "boolean");
+                            } else {
+                                logger.warn(
+                                        "air-Q - airqHandler - getConfigData(): The air-Q data could not be extracted from this string: {}",
+                                        decEl);
+                            }
+                        }
+                    } else {
+                        logger.warn(
+                                "air-Q - airqHandler - getConfigData(): The air-Q data could not be extracted from this string: {}",
+                                ans);
+                    }
+                }
+            } catch (Exception e) {
+                logger.warn("air-Q - airqHandler - getConfigData(): Error in processConfigData(): {}", e.toString());
+            }
+        }
+    }
+
+    private void processType(JsonObject dec, String airqName, String channelName, String type) {
+        logger.trace("air-Q - airqHandler - processType(): airqName={}, channelName={}, type={}, dec={}", airqName,
+                channelName, type, dec);
+        if (dec.get(airqName) == null) {
+            logger.trace("air-Q - airqHandler - processType(): get({}) is null", airqName);
+            updateState(channelName, UnDefType.UNDEF);
+            if (type.contentEquals("pair")) {
+                updateState(channelName + "_maxerr", UnDefType.UNDEF);
+            }
+        } else {
+            switch (type) {
+                case "boolean":
+                    String itemval = dec.get(airqName).toString();
+                    if (itemval.contentEquals("true") || itemval.contentEquals("1")) {
+                        updateState(channelName, OnOffType.ON);
+                    } else if (itemval.contentEquals("false") || itemval.contentEquals("0")) {
+                        updateState(channelName, OnOffType.OFF);
+                    }
+                    logger.trace("air-Q - airqHandler - processType(): channel {} set to {}", channelName, itemval);
+                    break;
+                case "string":
+                case "time":
+                    String strstr = dec.get(airqName).toString();
+                    updateState(channelName, new StringType(strstr.substring(1, strstr.length() - 1)));
+                    logger.trace("air-Q - airqHandler - processType(): channel {} set to {}", channelName, strstr);
```

Comment:
All of these TRACE logs are not needed and are double ups as all changes to the channels state is logged automatically in the event.log
Even if you don't have TRACE level turned on, this will still use up CPU time checking if it is on or off.
As mentioned it is done multiple times here so please look at them all under this 1 comment, thanks.